test/rubygems/test_gem_bundler_version_finder.rb

```diff
@@ -78,6 +78,16 @@ def test_bundler_version_with_reason
     end
   end
 
+  def test_deleted_directory
+    require "tmpdir"
+
+    Dir.mktmpdir do |dir|
+      Dir.chdir(dir)
+    end
+
+    assert_nil bvf.bundler_version_with_reason
```

Comment:
I think this test probably should chdir back to its old dir, otherwise the other tests are going to have a hard time / fail in an order-dependent way.

Reply:
@bronzdoc can you elaborate on why you marked this as resolved? I have the same concern as @asf-stripe, but I may be missing some context. (E.g., does the setup process run before each test call `Dir.chdir`, or something like that?

Reply:
Thanks both of you! I fixed it. It will restore to the original PWD, somehow I thought I was passing that block to `Dir.chdir`...